Attempting to specify /root as a mountpoint fails

I have tried to add a separate partition mounted at `/root`:
```
image roothome.ubifs {
	ubifs {
		max-size = 2M
	}
	flashtype = w25q64
	mountpoint = /root
}
```
This produces an error:
```
INFO: cmd: "mkdir -p "/.../buildroot/output/build/genimage.tmp"" (stderr):
INFO: cmd: "rm -rf "/.../buildroot/output/build/genimage.tmp"/*" (stderr):
INFO: cmd: "mkdir -p "/.../buildroot/output/images"" (stderr):
INFO: cmd: "mkdir -p "/.../buildroot/output/build/genimage.tmp"" (stderr):
INFO: cmd: "cp -a "/.../buildroot/output/target" "/.../buildroot/output/build/genimage.tmp/root"" (stderr):
INFO: cmd: "mv "/.../buildroot/output/build/genimage.tmp/root/root" "/.../buildroot/output/build/genimage.tmp/root"" (stderr):
mv: '/.../buildroot/output/build/genimage.tmp/root/root' and '/.../buildroot/output/build/genimage.tmp/root/root' are the same file
make: *** [Makefile:815: target-post-image] Error 1
```

It appears genimage is copying the various partitions into the same folder as the root filesystem, which is contained in a folder named "root." This conflicts with the mountpoint name.

I think the best way to fix this (and not just kick the can to some other more obscure name) is to add another directory alongside:
```
genimage.tmp/
    root/ <-- the existing root directory
    mountpoints/ <-- new
        root/ <-- the overlay
```

--- a/genimage.c
+++ b/genimage.c
@@ -392,7 +392,7 @@ static struct mountpoint *add_mountpoint(const char *path)
 	path_sanitized = sanitize_path(path);
 	mp = xzalloc(sizeof(*mp));
 	mp->path = strdup(path);
-	xasprintf(&mp->mountpath, "%s/%s", tmppath(), path_sanitized);
+	xasprintf(&mp->mountpath, "%s/mp-%s", tmppath(), path_sanitized);
 	list_add_tail(&mp->list, &mountpoints);
 	free(path_sanitized);
 

